app/code/Magento/Cms/Model/Wysiwyg/Images/GetInsertImageContent.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Cms\Model\Wysiwyg\Images;
+
+use Magento\Catalog\Helper\Data;
+use Magento\Cms\Helper\Wysiwyg\Images as ImagesHelper;
+
+class GetInsertImageContent
+{
+    /**
+     * @var ImagesHelper
+     */
+    private $imagesHelper;
+
+    /**
+     * @var Data
+     */
+    private $catalogHelper;
+
+    /**
+     * PrepareImage constructor.
+     *
+     * @param ImagesHelper $imagesHelper
+     * @param Data $catalogHelper
+     */
+    public function __construct(
+        ImagesHelper $imagesHelper,
+        Data $catalogHelper
+    ) {
+        $this->imagesHelper = $imagesHelper;
+        $this->catalogHelper = $catalogHelper;
+    }
+
+    /**
+     * Prepare Image Contents for Insert
+     *
+     * @param string $encodedFilename
+     * @param bool $forceStaticPath
+     * @param bool $renderAsTag
+     * @param int|null $storeId
+     * @return string
+     */
+    public function execute(
+        string $encodedFilename,
+        bool $forceStaticPath,
+        bool $renderAsTag,
+        ?int $storeId = null
+    ): string {
+        $filename = $this->imagesHelper->idDecode($encodedFilename);
+
+        $this->catalogHelper->setStoreId($storeId);
```

Comment:
I don't see the reason to set stored to catalog helper here. Considering your investigation this call can be removed